lib/puppet/node/environment.rb

```diff
@@ -82,8 +82,12 @@ def self.remote(name)
   # @param name [Symbol] The environment name
   def initialize(name, modulepath, manifest, config_version)
     @name = name
-    @modulepath = modulepath
-    @manifest = manifest
+    @modulepath = modulepath.map { |p| Puppet::FileSystem.expanded_path_string(p) }
+    if manifest && !manifest.is_a?(Symbol) && Puppet::FileSystem.exist?(manifest)
+      @manifest = Puppet::FileSystem.expanded_path_string(manifest)
+    else
+      @manifest = manifest
+    end
```

Comment:
An environment is essential to initialize the loaders, their lifecycle is tied to the environment's.